src/ui/layer_list_panel.ts

```diff
@@ -338,11 +339,44 @@ export class LayerListPanel extends SidePanel {
     sidePanelManager: SidePanelManager,
     public manager: TopLevelLayerListSpecification,
     public state: LayerListPanelState,
+    public hideLayerPanel: TrackableBoolean,
+    public showLayerPanel?: TrackableBoolean,
   ) {
     super(sidePanelManager, state.location);
     const { itemContainer, layerDropZone } = this;
-    const { titleElement } = this.addTitleBar({ title: "" });
+    const { titleElement, titleBar } = this.addTitleBar({ title: "" });
     this.titleElement = titleElement!;
+
+    // Add layer panel toggle button to title bar
+    // Only show toggle button when showLayerPanel configuration is enabled
+    if (this.showLayerPanel?.value) {
+      const toggleButton = new CheckboxIcon(this.hideLayerPanel, {
+        svg: svg_eye,
+        backgroundScheme: "dark",
+        enableTitle: "Hide layer panel",
+        disableTitle: "Show layer panel",
+      });
+      toggleButton.element.style.order = "50"; // Position before close button (order: 100)
+      titleBar.appendChild(toggleButton.element);
+      this.registerDisposer(toggleButton);
+
+      // Watch for showLayerPanel configuration changes and hide/show button accordingly
+      this.registerDisposer(
+        this.showLayerPanel.changed.add(() => {
+          if (this.showLayerPanel!.value) {
+            // Configuration enabled - show button
+            if (!toggleButton.element.parentElement) {
+              titleBar.appendChild(toggleButton.element);
+            }
+          } else {
+            // Configuration disabled - hide button
+            if (toggleButton.element.parentElement) {
+              toggleButton.element.remove();
+            }
+          }
+        }),
```

Comment:
Could maybe simplify this check if removing the icon fully is not needed and instead hiding is sufficient
```suggestion
          toggleButton.element.style.display = this.showLayerPanel!.value
            ? ""
            : "none";
```